frontend/src/teacherPage/UpdateStudent.tsx

```diff
@@ -0,0 +1,74 @@
+import DialogTitle from "@mui/material/DialogTitle";
+import IconButton from "@mui/material/IconButton";
+import CloseIcon from "@mui/icons-material/Close";
+import DialogContent from "@mui/material/DialogContent";
+import Typography from "@mui/material/Typography";
+import DialogActions from "@mui/material/DialogActions";
+import {Button, Dialog} from "@mui/material";
+import React from "react";
+import {styled} from "@mui/material/styles";
+
+const BootstrapDialog = styled(Dialog)(({ theme }) => ({
+    '& .MuiDialogContent-root': {
+        padding: theme.spacing(2),
+    },
+    '& .MuiDialogActions-root': {
+        padding: theme.spacing(1),
+    },
+}));
+
+export type Props = {
+    open: boolean;
+    handleClose: () => void;
+    handleSave: () => void;
+    name: string;
+    surname: string;
+    changeName: (event: React.ChangeEvent<HTMLInputElement>) => void;
+    changeSurname: (event: React.ChangeEvent<HTMLInputElement>) => void;
+}
+export default function UpdateStudent(props: Props){
+
+    return(
+        <> <BootstrapDialog
+            onClose={props.handleClose}
+            aria-labelledby="customized-dialog-title"
+            open={props.open}
+        >
+            <DialogTitle sx={{ m: 0, p: 2 }} id="customized-dialog-title">
+                Update Student
+            </DialogTitle>
+            <IconButton
+                aria-label="close"
+                onClick={props.handleClose}
+                sx={{
+                    position: 'absolute',
+                    right: 8,
+                    top: 8,
+                    color: (theme) => theme.palette.grey[500],
+                }}
+            >
+                <CloseIcon />
+            </IconButton>
+            <DialogContent dividers>
+                <Typography gutterBottom>
+                    <label>
+                        Name:
+                        <input value={props.name} type="text"  onChange={props.changeName} />
+                    </label>
```

Comment:
I am not sure, if encapsulating input in a label-element is wise. Maybe try something like: 
```
<div>
    <label htmlFor={"name"}>Name:</label>
    <input id={"name"} type={"text"} onChange={props.changName}/>
</div>
```

Reply:
This is working. :)